proto/backend.proto

```diff
@@ -119,7 +119,8 @@ message CallResource {
 
 service Diagnostics {
   rpc CollectMetrics(CollectMetrics.Request) returns (CollectMetrics.Response);
-  rpc CheckHealth(CheckHealth.Request) returns (CheckHealth.Response);
+  rpc CheckPluginHealth(CheckHealth.PluginRequest) returns (CheckHealth.Response);
+  rpc CheckDatasourceHealth(CheckHealth.DatasourceRequest) returns (CheckHealth.Response);
```

Comment:
I was thinking leaving the CheckHealth rpc method without adding a new one and just provide PluginConfig in the request since it includes DataSourceConfig. I was planning to do the same for CollectMetrics. Too keep the protocol simple and it's the same approach as we've done for resources. We can still expose CheckPluginHealth/CheckDatasourceHealth in the SDK if we want to.

Reply:
When grafana-server wants to issue an healthcheck for the plugin datasourceConfig should be nil I guess?

When grafana-server wants to issue and healthcheck for an datasource it should be set?

Seems weird IMO. Why is datasourceconfig a field on pluginconfig? I must have missed that.

Reply:
To make the API applicable for both app and data source plugins. See resource example:
https://github.com/grafana/grafana/blob/9d858220ce6193baf2f131708d1c2c26691b8a94/pkg/plugins/backendplugin/backend_plugin.go#L217-L252

Reply:
I probably miss a lot of context and history for the resource call. But in this case, I would find it weird to only have one endpoint taking `PluginConfig`. We could solve this in the SDK and grafana server but I dont see why we shouldn't solve it at the protobuf layer. It makes the API's clear and concise.

Reply:
Ok, I didn't realize that parts of the info need to make a datasource request is in [datasource config](https://github.com/grafana/grafana-plugin-sdk-go/blob/master/proto/backend.proto#L10) and parts are in [plugin config](https://github.com/grafana/grafana-plugin-sdk-go/blob/master/proto/backend.proto#L20). I find this quite confusing TBH.